cryptoki/src/session/slot_token_management.rs

```diff
@@ -6,30 +6,31 @@ use crate::error::{Result, Rv};
 use crate::session::Session;
 use std::convert::TryInto;
 
-// See public docs on stub in parent mod.rs
-#[inline(always)]
-pub(super) fn init_pin(session: &Session, pin: &str) -> Result<()> {
-    unsafe {
-        Rv::from(get_pkcs11!(session.client(), C_InitPIN)(
-            session.handle(),
-            pin.as_ptr() as *mut u8,
-            pin.len().try_into()?,
-        ))
-        .into_result()
+impl Session {
+    /// Initialize the normal user's pin for a token
+    pub fn init_pin(&self, pin: &str) -> Result<()> {
+        unsafe {
+            Rv::from(get_pkcs11!(self.client(), C_InitPIN)(
+                self.handle(),
+                pin.as_ptr() as *mut u8,
+                pin.len().try_into()?,
+            ))
+            .into_result()
+        }
     }
-}
 
-// See public docs on stub in parent mod.rs
-#[inline(always)]
-pub(super) fn set_pin(session: &Session, old_pin: &str, new_pin: &str) -> Result<()> {
-    unsafe {
-        Rv::from(get_pkcs11!(session.client(), C_SetPIN)(
-            session.handle(),
-            old_pin.as_ptr() as *mut u8,
-            old_pin.len().try_into()?,
-            new_pin.as_ptr() as *mut u8,
-            new_pin.len().try_into()?,
-        ))
-        .into_result()
+    /// Changes the PIN of either the currently logged in user or of the `CKU_USER` if no user is
+    /// logged in.
+    pub fn set_pin(&self, old_pin: &str, new_pin: &str) -> Result<()> {
```

Comment:
>      which one exactly went from pub(super) to pub?

If my diff-eyes don't deceive me this is `pub` now but was `pub(super) fn set_pin` previously...?

Reply:
https://github.com/parallaxsecond/rust-cryptoki/blob/9934be7902a3a5b143d33240fdc8c250fa1243ce/cryptoki/src/session/mod.rs#L82

Reply:
(again, the method implementation stays in the submodule, but the external interface moved as I tried to explain here:
https://github.com/parallaxsecond/rust-cryptoki/pull/118#issuecomment-1382122686)

Reply:
Ah, right :facepalm: good point!